network.go

```diff
@@ -52,6 +52,8 @@ type NetworkController interface {
 	// Create a new network. The options parameter carry driver specific options.
 	// Labels support will be added in the near future.
 	NewNetwork(d *NetworkDriver, name string, options interface{}) (Network, error)
+	// Register an external driver.
+	RegisterDriver(name string, driver driverapi.Driver)
```

Comment:
@squaremo IMO, plugins depending on libnetwork is not correct and docker/docker depending on driverapi is also not correct. The only dependency for docker/docker should be on libnetwork. And to me, libnetwork depending on plugin infra to resolve remote drivers is actually fine.